### ISSUE TITLE
Fails to connect to Nintendo Switch API

### ISSUE THREAD

**Issue by CatsLover2006** [samuelthomas2774/nxapi#17]
Working one day, then I get this log (in the console) the next:
```
Last login: Tue Aug 23 11:42:05 on ttys001
/Users/chancelp/Downloads/Nintendo\ Switch\ Online.app/Contents/MacOS/Nintendo\ Switch\ Online ; exit;
chancelp@Chances-M1-MacBook-Air ~ % /Users/chancelp/Downloads/Nintendo\ Switch\ Online.app/Contents/MacOS/Nintendo\ Switch\ Online ; exit;
objc[10408]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x7ffa46e8d980) and /Users/chancelp/Downloads/Nintendo Switch Online.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libGLESv2.dylib (0x110386ce0). One of the two will be used. Which one is undefined.
Authenticating to Nintendo Switch Online app
Authenticating to Nintendo Switch Online app
(node:10406) UnhandledPromiseRejectionWarning: ReferenceError: __filename is not defined
(Use `Nintendo Switch Online --trace-warnings ...` to show where the warning was created)
(node:10406) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:10406) UnhandledPromiseRejectionWarning: ReferenceError: __filename is not defined
(node:10406) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
(node:10406) UnhandledPromiseRejectionWarning: ReferenceError: __filename is not defined
(node:10406) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 5)
(node:10406) UnhandledPromiseRejectionWarning: ReferenceError: __filename is not defined
(node:10406) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 6)
(node:10406) UnhandledPromiseRejectionWarning: ReferenceError: __filename is not defined
(node:10406) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 7)
^C
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```
This is running on macOS Ventura (M1 MBA, build 22A5321d) and splatnet2statink runs just fine under the same curcumstances.
This was running just fine before the update

**Comment by samuelthomas2774** [comment]
Nintendo updated their API earlier today to properly validate something currently I don't think anyone is sending properly. I think I've figured out how to fix this properly but you'll have to wait for an update for now. (If you're interested there's more technical details about this on Discord, which I'll add to #10 later.)

> splatnet2statink runs just fine under the same curcumstances

SplatNet 2 session tokens can be valid for a long time so it's probably using a cached session. If you remove the iksm_session from your config.txt or let it expire it should fail to re-authenticate automatically.

---

> ```
> ReferenceError: __filename is not defined
> ```

I set a reminder to fix this a while ago and completely forgot about it :sweat_smile:. It's caused when an error is thrown in the main process in the packaged version of the app, so this hides the original error, which in this case is that the authentication attempt was rejected. If you run the app with `DEBUG=*` it'll show more information (there should be a "got f parameter" message just before this, but still not the original error). (https://github.com/samuelthomas2774/nxapi/commit/223faab510dec12ae53c1717babfc6e5a7d6a018)
